test-utils/test/shared/setupRerender.test.js

```diff
@@ -0,0 +1,40 @@
+import { options, createElement as h, render } from 'preact';
+import { spy } from 'sinon';
+import { useState } from 'preact/hooks';
+
+import { setupScratch, teardown } from '../../../test/_util/helpers';
+import { setupRerender } from '../../src';
+
+/** @jsx h */
+describe('setupRerender', () => {
+
+	/** @type {HTMLDivElement} */
+	let scratch;
+
+	let rerender;
+
+	beforeEach(() => {
+		rerender = setupRerender();
+		scratch = setupScratch();
+	});
+
+	afterEach(() => {
+		teardown(scratch);
+	});
+
+	it('should reset options after rerender', () => {
+		expect(options.debounceRendering).to.equal(undefined);
+		rerender();
+		expect(options.debounceRendering).to.equal(undefined);
+	});
+
+	it('should restore options.debounceRendering', () => {
+		const spy = sinon.spy();
+
+		options.debounceRendering = spy;
+		rerender();
```

Comment:
I think the `setupRerender` call should happen here right after setting `options.debounceRendering`. Without that the test doesn't really test what it says it does.

```
1. act -> setupRerender sets options.debounceRendering
2. Test overwrites options.debounceRendering with a spy
```

What should be tested instead:

```
1. Set options.debounceRendering with a spy
2. act -> setupRerender overwrites options.debounceRendering
3. when act is done, options.debounceRendering should be a spy again.
```

Reply:
As far as I know previous debouce was historically not being reset by `setupRerender` so I don't really get it sorry.